### PR TITLE
Fix tests 10_module Suite Full

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | - Fix selectors <br/> - Add wait in switch window to allow page loading <br/> - Replace some pause with waitForVisible <br/> - Add string-length in some selectors (selenium get text from selector before loading it (ajax request))
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | - TEST_PATH=full/10_module/5_check_addons_catalog_page npm run specific-test -- --URL=shopURL <br/> - TEST_PATH=full/10_module/6_check_notifications npm run specific-test -- --URL=shopURL <br/> - TEST_PATH=full/10_module/7_discover_module npm run specific-test -- --URL=shopURL

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13568)

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi,

Looks like the detection of the new tab with Amazon Marketplace did not pass (although the tab was opened).

```
$ TEST_PATH=full/10_module/7_discover_module npm run specific-test -- --URL=http://ip6-localhost/PrestaShop/ --DIR=~/Télechargements -

> END2END@1.0.0 specific-test /home/thomas/Documents/github/PrestaShop/tests/E2E
> node_modules/mocha/bin/mocha test/campaigns/$TEST_PATH "--URL=http://ip6-localhost/PrestaShop/" "--DIR=~/Télechargements"



  Discover "Amazon market place" module in Back Office
    Login in the Back Office
      ✓ should open the browser
      ✓ should login successfully in the Back Office
    Check then install "ps_mbo" module
      ✓ should go to the "Dashboard" page (24654ms)
      ✓ should install the "MBO Module" if not installed
    Check that the "Amazon market place" module is opened
      ✓ should go to "Modules Catalog" page
      ✓ should search for the module "Amazon Market Place"
      ✓ should click on "Discover" button
      1) should verify it opens the addons Amazon market place product page in a new tab
    Check then uninstall "ps_mbo" module
      2) should go to "Module Catalog" page
      3) should go to "Module Manager" page
      4) should click on "Installed Modules" tab
      5) should search for ps_mbo module in the installed module tab
      6) should click on "Search" button
      7) should check if the module ps_mbo was installed
      ✓ should click on "Uninstall" button
      8) should go to the "Dashboard" page
    Check that the "Amazon market place" module is opened
      9) should go to "Modules Catalog" page

[...]
```

**Comment by boubkerbribri** [comment]
That why I add pause in switchWindow and worked for me
`.then(() => client.switchWindow(id,1000))`
We can't wait for selector because page is not loaded.
You can try to use `.then(() => client.switchWindow(id,2000))` to see if it work for you

File: tests/E2E/test/campaigns/common_scenarios/module.js [hunk 1]
```diff
@@ -336,7 +336,7 @@ module.exports = {
     test('should click on "Discover" button', () => client.waitForExistAndClick(ModulePage.discover_amazon_module_button));
     test('should verify it opens the addons Amazon market place product page in a new tab', () => {
       return promise
-        .then(() => client.switchWindow(id))
+        .then(() => client.switchWindow(id,1000))
         .then(() => client.refresh()) /**Adding refreshing page because sometimes is not well opened we have to refresh it before */
         .then(() => client.checkTextValue(ModulePage.module_name, "Amazon Market Place Module", 'contain'))
```
Comment (Quetzacoalt91) [inline tests/E2E/test/campaigns/common_scenarios/module.js]:
```suggestion
        .then(() => client.checkTextValue(ModulePage.module_name, "Amazon Market Place", 'contain'))
```

The name can change depending on the current language

Comment (Quetzacoalt91) [inline tests/E2E/test/campaigns/common_scenarios/module.js]: It looks like this selector is defined in 
* tests/E2E/test/selectors/BO/addons_catalog_page.js
* tests/E2E/test/selectors/BO/module_page.js

for the exact same target but with different values. Can you please merge this two values and use the value in `addons_catalog_page`

**Comment by Quetzacoalt91** [comment]
On the second script, I have an issue when refreshing the page while an module action is still in progress:

```
TEST_PATH=full/10_module/6_check_notifications npm run specific-test -- --URL=http://ip6-localhost/PrestaShop/ --DIR=~/Télechargements

> END2END@1.0.0 specific-test /home/thomas/Documents/github/PrestaShop/tests/E2E
> node_modules/mocha/bin/mocha test/campaigns/$TEST_PATH "--URL=http://ip6-localhost/PrestaShop/" "--DIR=~/Télechargements" 



  Check notification module in the Back Office
    Login in the Back Office
      ✓ should open the browser
      ✓ should login successfully in the Back Office
    Check then install "ps_mbo" module
      ✓ should go to the "Dashboard" page
      ✓ should install the "MBO Module" if not installed (42590ms)
    Configure "Bank Transfer" module then check notifications
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Alerts" tab
      ✓ should click on "Configure" button for "ps_wirepayment"
      ✓ should set the "Account owner" input
      ✓ should set the "Account details" textarea
      ✓ should set the "Bank address" textarea
      ✓ should click on "Save" button
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Alerts" tab
      ✓ should check that the "Alerts number" is decremented with 1
      ✓ should check that the configured module is not visible in the "Alerts" tab
      ✓ should go to "Module Manager" page
      ✓ should click on "Installed Modules"
      ✓ should search for "Bank transfer" module in the installed module tab
      ✓ should click on "Search" button
      ✓ should click on module dropdown
      ✓ should click on "Reset" action
      ✓ should click on "Yes, reset it" button
      ✓ should check that the success alert message is well displayed
      ✓ should go to "Dashboard" page
    Check then uninstall "ps_mbo" module
      ✓ should go to "Module Catalog" page
      ✓ should go to "Module Manager" page
      ✓ should click on "Installed Modules" tab
      ✓ should search for ps_mbo module in the installed module tab
      ✓ should click on "Search" button
      ✓ should check if the module ps_mbo was installed
      ✓ should click on "Uninstall" button
      ✓ should go to the "Dashboard" page
    Configure "Bank Transfer" module then check notifications
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Alerts" tab
      ✓ should click on "Configure" button for "ps_wirepayment"
      ✓ should set the "Account owner" input
      ✓ should set the "Account details" textarea
      ✓ should set the "Bank address" textarea
      ✓ should click on "Save" button
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Alerts" tab
      ✓ should check that the "Alerts number" is decremented with 1
      ✓ should check that the configured module is not visible in the "Alerts" tab
    Check then install "ps_mbo" module
      ✓ should go to the "Dashboard" page
      ✓ should install the "MBO Module" if not installed (37494ms)
    Upgrade module then check notifications
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Updates" tab
      ✓ should click on "Upgrade" button for the first module if there is at least one module to update
      ✓ should check that the updated module is not visible in the "Updates" tab if there is at least one module to update (22796ms)
      1) should check that the "Updates number" is decremented with 1 if there is at least one module to update
    Check then uninstall "ps_mbo" module
      2) should go to "Module Catalog" page
      ✓ should go to "Module Manager" page (24328ms)
      ✓ should click on "Installed Modules" tab
      ✓ should search for ps_mbo module in the installed module tab
      ✓ should click on "Search" button
      ✓ should check if the module ps_mbo was installed
      ✓ should click on "Uninstall" button
      ✓ should go to the "Dashboard" page
    Upgrade module then check notifications
      ✓ should go to "Modules > Module Manager" page
      ✓ should click on "Updates" tab
      ✓ should click on "Upgrade" button for the first module if there is at least one module to update
      ✓ should check that the updated module is not visible in the "Updates" tab if there is at least one module to update
      ✓ should check that the "Updates number" is decremented with 1 if there is at least one module to update
    Reset "Bank Transfer" module
      ✓ should go to "Module Manager" page
      ✓ should click on "Installed Modules"
      ✓ should search for "Bank transfer" module in the installed module tab
      ✓ should click on "Search" button
      ✓ should click on module dropdown
      ✓ should click on "Reset" action
      ✓ should click on "Yes, reset it" button
      ✓ should check that the success alert message is well displayed
      ✓ should go to "Dashboard" page
    Logout from the Back Office
      ✓ should logout successfully from the Back Office


  71 passing (7m)
  2 failing

  1) Check notification module in the Back Office
       Upgrade module then check notifications
         should check that the "Updates number" is decremented with 1 if there is at least one module to update:
     Error: element ("//*[@id="subtab-AdminModulesUpdates"]//span[@class="notification-counter" and string-length(text())>0]") still not visible after 90000ms
      at new WaitUntilTimeoutError (node_modules/webdriverio/build/lib/utils/ErrorHandler.js:149:12)
      at /home/thomas/Documents/github/PrestaShop/tests/E2E/node_modules/webdriverio/build/lib/commands/waitUntil.js:29:19
      at <anonymous>

  2) Check notification module in the Back Office
       Check then uninstall "ps_mbo" module
         should go to "Module Catalog" page:
     Error: An element could not be located on the page using the given search parameters ("//*[@id="subtab-AdminParentModulesSf"]/a/..").
      at elements("//*[@id="subtab-AdminParentModulesSf"]/a/..") - getAttribute.js:25:17
      at getAttribute("//*[@id="subtab-AdminParentModulesSf"]/a/..", "class") - common.webdriverio.js:151:8



[mochawesome] Report JSON saved to /home/thomas/Documents/github/PrestaShop/tests/E2E/mochawesome-report/mochawesome.json

[mochawesome] Report HTML saved to /home/thomas/Documents/github/PrestaShop/tests/E2E/mochawesome-report/mochawesome.html


```

It appears an HTTP 500 is thrown when you refresh the page too quickly. We should wait for the ajax request to end first.

File: tests/E2E/test/campaigns/common_scenarios/module.js [hunk 2]
```diff
@@ -336,19 +336,18 @@ module.exports = {
     test('should click on "Discover" button', () => client.waitForExistAndClick(ModulePage.discover_amazon_module_button));
     test('should verify it opens the addons Amazon market place product page in a new tab', () => {
       return promise
-        .then(() => client.switchWindow(id))
+        .then(() => client.switchWindow(id,1000))
```
Comment (PierreRambaud) [inline tests/E2E/test/campaigns/common_scenarios/module.js]:
```suggestion
        .then(() => client.switchWindow(id, 1000))
```

File: tests/E2E/test/campaigns/common_scenarios/module.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/E2E/test/campaigns/common_scenarios/module.js]: Resolved
you should test with lang = en and country = fr

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @boubkerbribri

**Pull request merged**